pkg/apis/core/validation/shoot_test.go

```diff
@@ -419,6 +419,39 @@ var _ = Describe("Shoot Validation Tests", func() {
 					Expect(ValidateShootHAConfigUpdate(newShoot, shoot)).To(BeEmpty())
 				})
 			})
+
+			Context("Shoot is hibernated", func() {
+				It("Should not allow upgrading from non-HA to HA when Spec.Hibernation.Enabled is set to `true`", func() {
+					shoot.Spec.ControlPlane = &core.ControlPlane{}
+					ValueTrue := true
```

Comment:
Any reason to make this variable exported?
```suggestion
					valueTrue := true
```